### ISSUE TITLE
`stack ghci` omits `-i` flag if `hs-source-dirs` is `./.`

### ISSUE THREAD

**Issue by gelisam** [commercialhaskell/stack#3013]
### General summary/comments

Normally, `stack ghci` launches `ghci` by passing it a bunch of `-i` arguments specifying where to find the source of all the modules which `stack ghci` loads on startup. With `hs-source-dirs: src`, for example, stack passes `-i/.../src` to ghci, but with `hs-source-dirs: ./.`, it does not pass a corresponding flag. This is usually benign since `ghci` automatically looks at the current folder, but in a multi-package, the current folder is a folder containing the sub-package, not the folder in which the sources of the sub-package are located.

Note that I'm using `./.` instead of `.` because `.` entries are [ignored](https://github.com/sol/hpack/issues/119) by Cabal.

### Steps to reproduce

I created a minimal multi-package project demonstrating the problem [here](https://github.com/gelisam/stack-bug). It's a `stack.yaml` which includes a sub-package with its own `stack.yaml` and its own `mypackage.cabal` file, and the latter contains `hs-source-dirs: ./.`. Clone the repo and type `stack ghci` in the multi-package folder.

### Expected

A ghci session in which the modules of my sub-package are loaded:

    $ stack ghci
    [...]
    [1 of 1] Compiling MyModule

### Actual

    $ stack ghci
    [...]
    error: module ‘MyModule’ is a package module

[Here](https://gist.github.com/gelisam/b9368a3cd2e669501ec52a25ceada3a3) is the verbose output.

### Stack version

```
$ stack --version
Version 1.3.2, Git revision 3f675146590da4f3edf768b89355f798229da2a5 (4395 commits) x86_64 hpack-0.15.0
```

### Method of installation

`curl -sSL https://get.haskellstack.org/ | sh`

**Comment by gelisam** [comment]
Just tried with stack HEAD, looks like the bug is already fixed, thanks!

**Comment by gelisam** [comment]
Hmm, and now it's broken again, still with stack HEAD. I didn't even `git pull`, it's the exact same version. I don't know what else changed. Mysterious!

**Comment by mgsloan** [comment]
Hmm, strange, I can't reproduce the issue with HEAD.  Let me know if you figure out what changed!

**Comment by gelisam** [comment]
So I tried to fix it myself, but none of the changes I was making to stack's source were being picked up. I eventually discovered that even though `which stack` was telling me that `~/.local/bin/stack` was the first `stack` executable in my path, running `stack ghci` was still running the `/usr/local/bin/stack` one which had the bug! 😖 Weird, but definitely no stack's fault. 

The bug really is fixed in stack HEAD, sorry for the noise!
